lib/Conversion/ATenToTCF/CoreOpConversionPatterns.cpp

```diff
@@ -0,0 +1,73 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "npcomp/Conversion/ATenToTCF/Patterns.h"
+
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/IR/PatternMatch.h"
+#include "npcomp/Dialect/ATen/IR/ATenDialect.h"
+#include "npcomp/Dialect/TCF/IR/TCFOps.h"
+
+using namespace mlir;
+using namespace mlir::NPCOMP;
+
+namespace {
+
+/// The ATen AddOp actually has three arguments:
+///   self, other, alpha
+/// Alpha is an integer that is multiplied by 'other' prior to adding.
+class ConvertATenAdd : public OpRewritePattern<aten::AddOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(aten::AddOp srcAddOp,
+                                PatternRewriter &rewriter) const override {
+    // Special case: Match when alpha is constant 1, which is the default,
+    // quite common and maps directly to a TCF add.
+    APInt alphaValue;
+    if (matchPattern(srcAddOp.alpha(), m_ConstantInt(&alphaValue)) &&
+        alphaValue.getZExtValue() == 1) {
+      rewriter.replaceOpWithNewOp<tcf::AddOp>(
+          srcAddOp, srcAddOp.getResult().getType(), srcAddOp.self(),
+          srcAddOp.other());
+      return success();
+    }
+
+    return rewriter.notifyMatchFailure(
+        srcAddOp, "aten.add to tcf.add currently only supports alpha == 1");
+  }
+};
+
+/// Common conversion template for true binary elementwise ops.
+/// This does not apply to the handful of "binary-ish" PyTorch ops that are
```

Comment:
Most folks (including myself) probably aren't familiar with this concept of "binary-ish". Maybe mention an example or two? Are you referring to e.g. add above that takes "alpha"?

Reply:
Rephrased.